data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostPrepareBeaconProposer.java

```diff
@@ -113,8 +114,10 @@ public void handleRequest(final RestApiRequest request) throws JsonProcessingExc
     if (!isProposerDefaultFeeRecipientDefined) {
       STATUS_LOG.warnMissingProposerDefaultFeeRecipientWithPreparedBeaconProposerBeingCalled();
     }
-    validatorDataProvider.prepareBeaconProposer(request.getRequestBody());
-    request.respondWithCode(SC_OK);
+    request.respondAsync(
+        validatorDataProvider
+            .prepareBeaconProposer(request.getRequestBody())
+            .thenApply(AsyncApiResponse::respondOk));
```

Comment:
probably worth handling the case when errors occur here so that the user won't get a 500 if it does occur... a few are handled but if there's specific exceptions this is where you'd want to be assessing them...

Reply:
thanks, good to know, this specific endpoint wasn't handling specific exceptions before and don't think there are any to include as of now